Sources/BridgeMiddleware/BridgeMiddleware.swift

```diff
@@ -1,28 +1,80 @@
 import Foundation
 import SwiftRex
 
-infix operator -->: KeyPathComposition
-precedencegroup KeyPathComposition {
+infix operator |>: ForwardApplication
+precedencegroup ForwardApplication {
     associativity: left
-    lowerThan: AssignmentPrecedence
+    lowerThan: TernaryPrecedence
+    higherThan: AssignmentPrecedence
 }
 
-public struct ArrowMap<In, Value, Out> {
-    public init(valueIn: KeyPath<In, Value?>, valueOut: @escaping (Value) -> Out?) {
-        self.valueIn = valueIn
-        self.valueOut = valueOut
+infix operator >>>: ForwardComposition
+infix operator >=>: ForwardComposition
+precedencegroup ForwardComposition {
+    associativity: left
+    lowerThan: TernaryPrecedence
+    higherThan: ForwardApplication
+}
+
+/// Terminal object composition. Makes a Void out of anything.
+/// It's used in a composition to ignore whatever value is given.
+public func ignore<A>(_ a: A) {
+    return ()
+}
+
+/// Apply a value into a function. Given a function from A to B, applying A will result in B.
+public func |> <A, B>(_ value: A, f: @escaping (A) -> B) -> B {
+    f(value)
+}
+
+/// To compose two functions when the input of the second function matches the result of the first.
+public func >>> <A, B, C>(f: @escaping (A) -> B, g: @escaping (B) -> C) -> (A) -> C {
```

Comment:
Would it make sense to have a common package for all the FP basics?

Reply:
teufelaudio/FoundationExtensions has some of this. The problem is about diamond dependency linking problems. :(

SwiftRex itself has the `<>` operator only for monoidal composition, others like `|>`, `>>>`, `>=>`, `<*>` could be there too, or moved to a common place, but not always when you need these operators you want to link SwiftRex, and I don't want SwiftRex to also have an external dependency. This is really hard to decide and to be honest I don't have an answer.

If only Xcode was linking SPM stuff properly...